### ISSUE TITLE
Dropped tick label for fixedformatter

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8779]
master but also present on 1.5.3
```
imshow(rand(5, 5)); gca().set_xticklabels(["foo", "bar"])
```
results in
![figure_1](https://user-images.githubusercontent.com/1322974/27292549-a77001c2-54c8-11e7-9f26-ad6d9fa02798.png)
Almost certain this is due to the xticks internally including a tick at the position "just left of bounds".  Seems related (not directly, though) to the issue discussed in #7042.

**Comment by tacaswell** [comment]
It is not safe to use `FixedFormatter` with a non-fixed locator.

**Comment by anntzer** [comment]
I agree it is not really suitable, but locators should really do an effort to return only ticks that are actually going to be drawn.

**Comment by anntzer** [comment]
xref #5560
